### PR TITLE
Panel: Add popup window button

### PR DESCRIPTION
**Pull request by yamgent**
**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [X] New feature

Resolves https://github.com/MarkBind/markbind/issues/225

**What is the rationale for this request?**
User wants a way to add popup windows for panels.

**What changes did you make? (Give an overview)**
* Add a popup window button to the `Panel`.
* Gave `Panel` a new attribute `popup-url`, which when specified, will be the URL that the popup window button navigates to.

**Provide a demo:**
popup

**Is there anything you'd like reviewers to focus on?**
The description of the new button in the documentation.

**Testing instructions:**
1. Navigate to the panel that has the popup window button enabled.
1. Click on the button. A new window should appear.

### CODE REVIEW

**Comment by damithc** [comment]
I guess you'll be adding a way to disable it too? we have `no-close` etc. for other buttons.

File: docs/example/accordionDocs.vue [hunk 1]
```diff
@@ -95,6 +95,20 @@
     <doc-code language="markup">
       <panel header="Dynamic Loading" src="/docs/loadContent.html#fragment" minimized></panel>
     </doc-code>
+    <br>
+
+    <h4>If <code>popup-url</code> attribute is provided, the panel shows a popup button, which when clicked, opens a new window to the specified url.</h4>
```
[the remark this answers is not in the document]
Comment (acjh) [reply docs/example/accordionDocs.vue]: It's actually still an (absolute, root-relative or relative) URL.

File: docs/example/accordionDocs.vue [hunk 2]
```diff
@@ -97,15 +97,15 @@
...
+    <h4>If <code>popup-url</code> attribute is provided, a popup button will be displayed. Upon clicking, it opens a new window to the specified url.</h4>
     <div class="bs-example">
       <panel header="Some panel" popup-url="/docs/loadContent.html">
         This panel has a popup.
       </panel>
     </div>
     <doc-code language="markup">
       <panel header="Some panel" popup-url="/docs/loadContent.html">
-        ...
+        <div class="."></div>
```
Comment (acjh) [inline docs/example/accordionDocs.vue]: Why?

Comment (yamgent) [reply docs/example/accordionDocs.vue]: Oops, my apologies, not intentional (not sure how it got there, WebStorm expanded it? :P)

**Comment by yamgent** [comment]
Minor error in the documentation fixed.

File: src/Panel.vue [hunk 3]
```diff
@@ -34,6 +34,10 @@
                                 @click.stop="close()">
                             <span class="glyphicon glyphicon-remove" aria-hidden="true"></span>
                         </button>
+                        <button type="button" class="popup-button btn btn-default"
+                                v-show="this.popupUrl !== null" @click.stop="openPopup()">
```
Comment (acjh) [inline src/Panel.vue]: `@click.stop` on new line.

File: docs/example/accordionDocs.vue [hunk 4]
```diff
@@ -226,6 +240,12 @@
         <p><code>false</code></p>
         <p>Whether to show the expand switch.</p>
       </div>
+      <div>
+        <p>popup-url</p>
+        <p><code>String</code></p>
+        <p></p>
+        <p>The url that the popup window will navigate to. The url can be absolute, relative or root-relative.</p>
+      </div>
```
Comment (acjh) [inline docs/example/accordionDocs.vue]: Move after `bottom-switch` for now (switch options are grouped).

File: docs/example/accordionDocs.vue [hunk 5]
```diff
@@ -95,6 +95,20 @@
     <doc-code language="markup">
       <panel header="Dynamic Loading" src="/docs/loadContent.html#fragment" minimized></panel>
     </doc-code>
+    <br>
+
+    <h4>If <code>popup-url</code> attribute is provided, a popup button will be displayed. Upon clicking, it opens a new window to the specified url.</h4>
+    <div class="bs-example">
+      <panel header="Some panel" popup-url="/docs/loadContent.html">
```
Comment (acjh) [inline docs/example/accordionDocs.vue]: Root-relative url likely won't work when published on GitHub. See 7dcb507.

File: docs/example/accordionDocs.vue [hunk 6]
```diff
@@ -95,6 +95,20 @@
...
+        This panel has a popup.
+      </panel>
+    </div>
+    <doc-code language="markup">
+      <panel header="Some panel" popup-url="/docs/loadContent.html">
+        ...
```
Comment (acjh) [inline docs/example/accordionDocs.vue]: For simple content, show as-is.

File: docs/example/accordionDocs.vue [hunk 7]
```diff
@@ -95,6 +95,20 @@
     <doc-code language="markup">
       <panel header="Dynamic Loading" src="/docs/loadContent.html#fragment" minimized></panel>
     </doc-code>
+    <br>
+
+    <h4>If <code>popup-url</code> attribute is provided, a popup button will be displayed. Upon clicking, it opens a new window to the specified url.</h4>
```
Comment (acjh) [inline docs/example/accordionDocs.vue]: How about:
- "will be `shown`" for consistent grammar in this docs.
- "If clicked, it opens the specified url in a new window."

**Pull request merged**